README.md

```diff
@@ -26,7 +26,9 @@ differences:
 
 ## Source Configuration
 
-* `repository`: *Required.* The name of the repository, e.g. `alpine`.
+* `repository`: *Required.* The name of the repository, e.g. `alpine`. If using ecr
+    you only need the repository name, not the full URI e.g. `alpine` not 
+    `012345678910.dkr.ecr.us-east-1.amazonaws.com/alpine`
```

Comment:
This is confusing I reckon. The [docker-image](https://github.com/concourse/docker-image-resource) resource also integrates with ECR but you are required to put the full repository name in.

Reply:
I was going to suggest that our code was better as it's less error prone.  The [code](https://github.com/dwp/registry-image-resource/blob/master/types.go#L112) uses the Auth data to automatically populate that repository endpoint.  However, because the [code](https://github.com/dwp/registry-image-resource/blob/master/types.go#L81-L83) also has the ability to switch role, I guess it's feasible that one could auth against one account, then assume a role that has permissions to pull from/push to registries in multiple other accounts.

Having looked at the API, I think one approach that could be taken here is to:

1. Use the default registry if no specific configuration is given to do anything else (i.e. the current behaviour); the docs could be improved here to make this behaviour more explicit
2. Have an `aws_source_registry` config item that takes a single string of the registry from which to pull a container image
3. Have an `aws_dest_registries` config item that takes a comma-separated list of strings specifying one or more registries into which the container image should be pushed

With all that said though, we currently have no use case for the above at all so I think this extension should really be tackled under its own issue/PR. The way we handle this is that our Concourse instance has an access key that is permitted to assume a role in each of our 2 accounts (dev & prod).  We have 1 pipeline consisting of 2 jobs, one for pushing to dev, and another one for prod. Once the assume role is done (via `aws_role_arn`) then the default registry is correct for that account/job. That enables us to more smoothly promote our containers through the environments too; rather than having a single pipeline/job that pushes to all account registries simultaneously, each job handles one registry and the container doesn't get pushed to prod before dev is confirmed as OK.